plugins/modules/foreman_provisioning_template.py

```diff
@@ -284,23 +272,21 @@ def main():
                 'snippet',
                 'user_data',
                 'ZTP',
-            ]),
+            ], type='entity', flat_name='template_kind_id'),
             template=dict(),
-            file_name=dict(type='path'),
-            locations=dict(type='list'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
             locked=dict(type='bool'),
             name=dict(),
-            organizations=dict(type='list'),
-            operatingsystems=dict(type='list'),
-            state=dict(default='present', choices=['absent', 'present_with_defaults', 'present']),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+            operatingsystems=dict(type='entity_list', flat_name='operatingsystem_ids'),
+            snippet=dict(type='invisible'),
```

Comment:
Why invisible? What does it mean? Is read-only perhaps a better name?

Reply:
My understanding is that "invisible" means a param we pass to Foreman, but that is not exposed in the Ansible module arguments.

Reply:
Do we document the entity spec for module authors somewhere?

Reply:
Not yet. I wrote that on @mdellweg's TODO the other day ;)

Reply:
We should at some point. At the moment it's like look at foreman_domain and start copy and paste from there.

Reply:
https://github.com/theforeman/foreman-ansible-modules/pull/385/commits/f9fe9f82425ee661bdd9efbf08d98fedadd7097d?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8